### PR TITLE
add shared module metadata

### PR DESCRIPTION
**Pull request by DwayneJengSage**

### CODE REVIEW

File: bridge-api/definitions/shared_module_metadata.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+properties:
+    id:
+        type: string
+        description: Unique ID of the module.
+    licenseRestricted:
+        type: boolean
+        description: True if usage of this module is restricted by license or copyright.
+    moduleType:
+        $ref: ./enums/shared_module_type.yml
+        readOnly: true
```
Comment (alxdarksage) [inline bridge-api/definitions/shared_module_metadata.yml:16]: You won't be able to set this in the object if you're constructing it, is that ok?

**Review by alxdarksage — APPROVED** [review]
See comment though about one field that I don't think will have a setter.

File: bridge-api/definitions/shared_module_metadata.yml (hunk 1 above)
Comment (DwayneJengSage) [reply bridge-api/definitions/shared_module_metadata.yml:16]: It should be okay. On the server-side, moduleType is actually inferred from the presence/absence of schemaId and surveyGuid, so the client shouldn't be able to set it anyway.

Ironically, the readOnly flag is actually ignored by swagger for whatever reason, and there *is* in fact a setter for it on the JavaSDK...

Comment (alxdarksage) [reply bridge-api/definitions/shared_module_metadata.yml:16]: Okay, looks good to merge

**Pull request merged**